front/src/applications/stdcm/utils/__tests__/simulationReportSheet.spec.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  generateCodeNumber,
+  formatCreationDate,
+  getStopDurationTime,
+  extractSpeedLimit,
+  computeStopDepartureTime,
+  addMinutesToTime,
+  getStopDurationBetweenTwoPositions,
+} from 'applications/stdcm/utils/formatSimulationReportSheet';
+
+describe('generateCodeNumber', () => {
+  it('should return a formatted string', () => {
+    const codeNumber = generateCodeNumber();
+    expect(codeNumber).toMatch(/^\d{2}\d{2}-\d{3}-\d{3}$/);
+  });
+});
+
+// TODO DROP STDCM V1
+describe('formatCreationDate', () => {
+  it('should format the date correctly', () => {
+    const departure_time = '2024-07-08T10:20:30Z';
+    const formattedDate = formatCreationDate(departure_time);
+    expect(formattedDate).toEqual({
+      day: '08',
+      month: '07',
+      year: 2024,
+      hours: 10,
+      minutes: 20,
+    });
+  });
+});
+
+describe('getStopDurationTime', () => {
+  it('should return correct time format', () => {
+    expect(getStopDurationTime(30)).toBe('30 sec');
+    expect(getStopDurationTime(120)).toBe('2 min');
+  });
+});
+
+describe('extractSpeedLimit', () => {
+  it('should extract the speed limit correctly', () => {
+    const speedLimitByTag1 = 'Voyageurs 100 - MA100';
+    const speedLimitByTag2 = 'Voyageurs';
+    expect(extractSpeedLimit(speedLimitByTag1)).toBe('MA100');
+    expect(extractSpeedLimit(speedLimitByTag2)).toBe('Voyageurs');
+  });
+});
+
+describe('computeStopDepartureTime', () => {
+  it('should compute stop departure time correctly', () => {
+    expect(computeStopDepartureTime('10:30', '15:00')).toBe('10:45');
+    expect(computeStopDepartureTime('23:59', '01:00')).toBe('00:00');
+    expect(computeStopDepartureTime('23:59', '02:00')).toBe('00:01');
+  });
```

Comment:
It's not really important here but for your future tests, usually, you would do something like that (one except for each `it`). This way we know why you do this test specifically. But don't bother to change all of them, it's fine.

```suggestion
  it('should compute stop departure time correctly for a standard time', () => {
    expect(computeStopDepartureTime('10:30', '15:00')).toBe('10:45');
  });
  it('should compute stop departure time correctly when changing days', () => {
    expect(computeStopDepartureTime('23:59', '01:00')).toBe('00:00');
  });
  etc
```

Reply:
Thank you for the tips ! done